tjpcov/covariance_io.py

```diff
@@ -81,43 +79,46 @@ def create_sacc_cov(self, cov, output="cls_cov.fits", overwrite=False):
             :obj:`sacc.sacc.Sacc`: The final sacc file with the covariance
             matrix included.
         """
-        output = os.path.join(self.get_outdir(), output)
+        output_file_nm = os.path.join(self.outdir, output)
 
-        s = self.get_sacc_file().copy()
-        s.add_covariance(cov, overwrite=True)
+        sacc_clone = self.get_sacc_file().copy()
+        sacc_clone.add_covariance(cov, overwrite=True)
 
-        if os.path.isfile(output) and (not overwrite):
-            date = datetime.utcnow()
-            timestamp = date.strftime("%Y%m%d%H%M%S")
-            output_new = output + f"_{timestamp}"
-            warnings.warn(
-                f"Output file {output} already exists. "
-                "Appending the UTC time to the filename to avoid "
-                "losing the covariance computation. Writing sacc "
-                "file to {output_new}"
-            )
-            output = output_new
+        if not os.path.isfile(output_file_nm) or overwrite:
+            sacc_clone.save_fits(output_file_nm, overwrite=overwrite)
+            return sacc_clone
 
-        s.save_fits(output, overwrite=overwrite)
+        timestamp = datetime.utcnow().strftime("%Y%m%d%H%M%S")
+        output_file_nm = output_file_nm + f"_{timestamp}"
+        warnings.warn(
+            f"Output file {output} already exists. "
+            "Appending the UTC time to the filename to avoid "
+            "losing the covariance computation. Writing sacc "
+            f"file to {output_file_nm}"
+        )
 
-        return s
+        sacc_clone.save_fits(output_file_nm, overwrite=overwrite)
```

Comment:
Hmm, not sure if not returning could break the integration with TXPipe. If you think it's better not returning it, I'd check what TXPipe is doing, otherwise I'd just add a TODO or sth like that.
In my opinion, I think it's convenient to get it returned there.

Reply:
Ah excellent call.  I meant to be more careful about changing existing functionality.  I will check and make sure nothing is breaking other tools.